dotcom-rendering/scripts/test-data/gen-fixtures.js

```diff
@@ -124,171 +124,209 @@ const articles = [
+				null,
+				4,
+			)}`;
+			return fs.writeFile(
+				`${root}/fixtures/generated/match-report.ts`,
+				contents,
+				'utf8',
+			);
+		})
+		.then(() => 'match-report.ts')
+		.catch((err) => {
+			throw new Error('Failed to create match-report.ts', { cause: err });
+		}),
+);
 
-	// Story package
-	requests.push(
-		fetch(
-			'https://api.nextgen.guardianapps.co.uk/story-package/science/2021/feb/18/life-on-mars-nasa-keeps-looking-with-perseverance-rover-mission.json?dcr=true',
-		)
-			.then((res) => res.json())
-			.then((json) => {
-				// Write the new fixture data
-				const contents = `${HEADER}export const storyPackage = ${JSON.stringify(
-					json,
-					null,
-					4,
-				)}`;
-				fs.writeFileSync(
-					`${root}/fixtures/generated/story-package.ts`,
-					contents,
-					'utf8',
-				);
-				console.log(`Created story-package.ts`);
-			}),
-	);
+// Series
+requests.push(
+	fetch(
+		'https://api.nextgen.guardianapps.co.uk/series/tv-and-radio/series/tv-review.json?dcr',
+	)
+		.then((res) => res.json())
+		.then((json) => {
+			// Write the new fixture data
+			const contents = `${HEADER}export const series = ${JSON.stringify(
+				json,
+				null,
+				4,
+			)}`;
+			return fs.writeFile(
+				`${root}/fixtures/generated/series.ts`,
+				contents,
+				'utf8',
+			);
+		})
+		.then(() => 'series.ts')
+		.catch((err) => {
+			throw new Error('Failed to create series.ts', { cause: err });
+		}),
+);
 
-	Promise.all(requests).finally(() => {
-		console.log('\nFormatting files...');
-		execa('prettier', [
-			'./fixtures/**/*',
-			'--write',
-			'--loglevel',
-			'error',
-		]).then(() => {
+// Story package
+requests.push(
+	fetch(
+		'https://api.nextgen.guardianapps.co.uk/story-package/science/2021/feb/18/life-on-mars-nasa-keeps-looking-with-perseverance-rover-mission.json?dcr=true',
+	)
+		.then((res) => res.json())
+		.then((json) => {
+			// Write the new fixture data
+			const contents = `${HEADER}export const storyPackage = ${JSON.stringify(
+				json,
+				null,
+				4,
+			)}`;
+			return fs.writeFile(
+				`${root}/fixtures/generated/story-package.ts`,
+				contents,
+				'utf8',
+			);
+		})
+		.then(() => 'story-package.ts')
+		.catch((err) => {
+			throw new Error('Failed to create story-package.ts', {
+				cause: err,
+			});
+		}),
+);
+
+Promise.allSettled(requests)
+	.then((results) => {
+		const successful = results.filter(
+			({ status }) => status === 'fulfilled',
+		);
+		const failed = results.filter(({ status }) => status === 'rejected');
+
+		if (successful.length > 0) {
 			console.log(
-				`\nDone ✅ Successfully created ${requests.length} fixtures\n`,
+				`\n✅ Successfully created ${successful.length} / ${requests.length} fixtures:\n`,
+				...successful.map(({ value }) => `${value}\n`),
 			);
-		});
+		}
+
+		if (failed.length > 0) {
+			console.error(
+				`❌ Failed to create ${failed.length} / ${requests.length} fixtures, with errors:\n`,
+				...failed.flatMap(({ reason }) => [reason, `\n`]),
```

Comment:
Why use both `map` + `${value}\n` and `flatMap` + `[value, "\n"]`?

Seems like picking one approach and sticking with it would be more consistent?

Reply:
The implementations are slightly different because the requirements are slightly different.

In the first case the `value`s are just `string`s, so all that's needed is to add a newline and pass them directly to the console method for printing. That's `string => string`, so `map` can handle that.

In the second case the `reason`s are error objects. If you put them in a template string to add the newline it implicitly calls `toString` on them and that only includes the error message. What I wanted to do instead was to pass the error object to the console method directly, because it prints out the full object with stack trace, cause etc. I also still wanted to print a newline. That's `string => [Error, string]`, so I needed to step up to `flatMap`.

I could also have written a formatter function, `Error => string`, to build the different parts of an error message and a newline into a `string` for `map`, but this seemed simpler.